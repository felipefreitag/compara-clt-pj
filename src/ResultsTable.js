import React from 'react';
import { Table, TableBody, TableHeader } from 'material-ui/Table';
import { TableHeaderColumn, TableRow, TableRowColumn } from 'material-ui/Table';
import { translate, omit } from './functions'

const ResultsTable = ({ toShow }) => {
  const noName = omit(toShow, 'name');

  return (
    <Table style={{ margin: '0 1rem', width: '90%' }}>
      <TableHeader
        displaySelectAll={false}
        adjustForCheckbox={false}
      >
        <TableRow>
          <TableHeaderColumn>
            Resultados { toShow && translate(toShow.name) }
          </TableHeaderColumn>
        </TableRow>
      </TableHeader>
      <TableBody displayRowCheckbox={false}>
        {Object.keys(noName).map((key, index) => {
            return (
              <TableRow key={index}>
                <TableRowColumn>
                  {translate(key)}
                </TableRowColumn>
                <TableRowColumn style={{ textAlign: 'right' }}>
                  R$ {toShow[key]}
                </TableRowColumn>
              </TableRow>
            )
          }
        )}
      </TableBody>
    </Table>
  );
}

export default ResultsTable
