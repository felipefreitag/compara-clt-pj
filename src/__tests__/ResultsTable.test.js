import React from 'react'
import Enzyme, { shallow } from 'enzyme'
import Adapter from 'enzyme-adapter-react-16'
import ResultsTable from '../ResultsTable'

Enzyme.configure({ adapter: new Adapter() })

it('renders correctly', () => {
  const wrapper = shallow(<ResultsTable toShow={{ foo: 'bar' }}/>)
  expect(wrapper).toMatchSnapshot()
})
