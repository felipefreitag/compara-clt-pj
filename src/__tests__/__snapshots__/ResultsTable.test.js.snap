// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <ResultsTable
    toShow={
        Object {
            "foo": "bar",
          }
    }
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": null,
    "nodeType": "class",
    "props": Object {
      "allRowsSelected": false,
      "children": Array [
        <TableHeader
          adjustForCheckbox={false}
          displaySelectAll={false}
          enableSelectAll={true}
          selectAllSelected={false}
>
          <TableRow
                    displayBorder={true}
                    hoverable={false}
                    hovered={false}
                    selectable={true}
                    striped={false}
          >
                    <TableHeaderColumn>
                              Resultados 
                    </TableHeaderColumn>
          </TableRow>
</TableHeader>,
        <TableBody
          allRowsSelected={false}
          deselectOnClickaway={true}
          displayRowCheckbox={false}
          multiSelectable={false}
          preScanRows={true}
          selectable={true}
          style={Object {}}
>
          <TableRow
                    displayBorder={true}
                    hoverable={false}
                    hovered={false}
                    selectable={true}
                    striped={false}
          >
                    <TableRowColumn
                              hoverable={false}
                    >
                              foo
                    </TableRowColumn>
                    <TableRowColumn
                              hoverable={false}
                              style={
                                        Object {
                                                  "textAlign": "right",
                                                }
                              }
                    >
                              R$ 
                              bar
                    </TableRowColumn>
          </TableRow>
</TableBody>,
      ],
      "fixedFooter": true,
      "fixedHeader": true,
      "height": "inherit",
      "multiSelectable": false,
      "selectable": true,
      "style": Object {
        "margin": "0 1rem",
        "width": "90%",
      },
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": null,
        "nodeType": "class",
        "props": Object {
          "adjustForCheckbox": false,
          "children": <TableRow
            displayBorder={true}
            hoverable={false}
            hovered={false}
            selectable={true}
            striped={false}
>
            <TableHeaderColumn>
                        Resultados 
            </TableHeaderColumn>
</TableRow>,
          "displaySelectAll": false,
          "enableSelectAll": true,
          "selectAllSelected": false,
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": null,
          "nodeType": "class",
          "props": Object {
            "children": <TableHeaderColumn>
              Resultados 
</TableHeaderColumn>,
            "displayBorder": true,
            "hoverable": false,
            "hovered": false,
            "selectable": true,
            "striped": false,
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": null,
            "nodeType": "class",
            "props": Object {
              "children": Array [
                "Resultados ",
                undefined,
              ],
            },
            "ref": null,
            "rendered": Array [
              "Resultados ",
              undefined,
            ],
            "type": [Function],
          },
          "type": [Function],
        },
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": null,
        "nodeType": "class",
        "props": Object {
          "allRowsSelected": false,
          "children": Array [
            <TableRow
              displayBorder={true}
              hoverable={false}
              hovered={false}
              selectable={true}
              striped={false}
>
              <TableRowColumn
                            hoverable={false}
              >
                            foo
              </TableRowColumn>
              <TableRowColumn
                            hoverable={false}
                            style={
                                          Object {
                                                        "textAlign": "right",
                                                      }
                            }
              >
                            R$ 
                            bar
              </TableRowColumn>
</TableRow>,
          ],
          "deselectOnClickaway": true,
          "displayRowCheckbox": false,
          "multiSelectable": false,
          "preScanRows": true,
          "selectable": true,
          "style": Object {},
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": "0",
            "nodeType": "class",
            "props": Object {
              "children": Array [
                <TableRowColumn
                  hoverable={false}
>
                  foo
</TableRowColumn>,
                <TableRowColumn
                  hoverable={false}
                  style={
                                    Object {
                                                      "textAlign": "right",
                                                    }
                  }
>
                  R$ 
                  bar
</TableRowColumn>,
              ],
              "displayBorder": true,
              "hoverable": false,
              "hovered": false,
              "selectable": true,
              "striped": false,
            },
            "ref": null,
            "rendered": Array [
              Object {
                "instance": null,
                "key": null,
                "nodeType": "class",
                "props": Object {
                  "children": "foo",
                  "hoverable": false,
                },
                "ref": null,
                "rendered": "foo",
                "type": [Function],
              },
              Object {
                "instance": null,
                "key": null,
                "nodeType": "class",
                "props": Object {
                  "children": Array [
                    "R$ ",
                    "bar",
                  ],
                  "hoverable": false,
                  "style": Object {
                    "textAlign": "right",
                  },
                },
                "ref": null,
                "rendered": Array [
                  "R$ ",
                  "bar",
                ],
                "type": [Function],
              },
            ],
            "type": [Function],
          },
        ],
        "type": [Function],
      },
    ],
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": null,
      "nodeType": "class",
      "props": Object {
        "allRowsSelected": false,
        "children": Array [
          <TableHeader
            adjustForCheckbox={false}
            displaySelectAll={false}
            enableSelectAll={true}
            selectAllSelected={false}
>
            <TableRow
                        displayBorder={true}
                        hoverable={false}
                        hovered={false}
                        selectable={true}
                        striped={false}
            >
                        <TableHeaderColumn>
                                    Resultados 
                        </TableHeaderColumn>
            </TableRow>
</TableHeader>,
          <TableBody
            allRowsSelected={false}
            deselectOnClickaway={true}
            displayRowCheckbox={false}
            multiSelectable={false}
            preScanRows={true}
            selectable={true}
            style={Object {}}
>
            <TableRow
                        displayBorder={true}
                        hoverable={false}
                        hovered={false}
                        selectable={true}
                        striped={false}
            >
                        <TableRowColumn
                                    hoverable={false}
                        >
                                    foo
                        </TableRowColumn>
                        <TableRowColumn
                                    hoverable={false}
                                    style={
                                                Object {
                                                            "textAlign": "right",
                                                          }
                                    }
                        >
                                    R$ 
                                    bar
                        </TableRowColumn>
            </TableRow>
</TableBody>,
        ],
        "fixedFooter": true,
        "fixedHeader": true,
        "height": "inherit",
        "multiSelectable": false,
        "selectable": true,
        "style": Object {
          "margin": "0 1rem",
          "width": "90%",
        },
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": null,
          "nodeType": "class",
          "props": Object {
            "adjustForCheckbox": false,
            "children": <TableRow
              displayBorder={true}
              hoverable={false}
              hovered={false}
              selectable={true}
              striped={false}
>
              <TableHeaderColumn>
                            Resultados 
              </TableHeaderColumn>
</TableRow>,
            "displaySelectAll": false,
            "enableSelectAll": true,
            "selectAllSelected": false,
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": null,
            "nodeType": "class",
            "props": Object {
              "children": <TableHeaderColumn>
                Resultados 
</TableHeaderColumn>,
              "displayBorder": true,
              "hoverable": false,
              "hovered": false,
              "selectable": true,
              "striped": false,
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": null,
              "nodeType": "class",
              "props": Object {
                "children": Array [
                  "Resultados ",
                  undefined,
                ],
              },
              "ref": null,
              "rendered": Array [
                "Resultados ",
                undefined,
              ],
              "type": [Function],
            },
            "type": [Function],
          },
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": null,
          "nodeType": "class",
          "props": Object {
            "allRowsSelected": false,
            "children": Array [
              <TableRow
                displayBorder={true}
                hoverable={false}
                hovered={false}
                selectable={true}
                striped={false}
>
                <TableRowColumn
                                hoverable={false}
                >
                                foo
                </TableRowColumn>
                <TableRowColumn
                                hoverable={false}
                                style={
                                                Object {
                                                                "textAlign": "right",
                                                              }
                                }
                >
                                R$ 
                                bar
                </TableRowColumn>
</TableRow>,
            ],
            "deselectOnClickaway": true,
            "displayRowCheckbox": false,
            "multiSelectable": false,
            "preScanRows": true,
            "selectable": true,
            "style": Object {},
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": "0",
              "nodeType": "class",
              "props": Object {
                "children": Array [
                  <TableRowColumn
                    hoverable={false}
>
                    foo
</TableRowColumn>,
                  <TableRowColumn
                    hoverable={false}
                    style={
                                        Object {
                                                            "textAlign": "right",
                                                          }
                    }
>
                    R$ 
                    bar
</TableRowColumn>,
                ],
                "displayBorder": true,
                "hoverable": false,
                "hovered": false,
                "selectable": true,
                "striped": false,
              },
              "ref": null,
              "rendered": Array [
                Object {
                  "instance": null,
                  "key": null,
                  "nodeType": "class",
                  "props": Object {
                    "children": "foo",
                    "hoverable": false,
                  },
                  "ref": null,
                  "rendered": "foo",
                  "type": [Function],
                },
                Object {
                  "instance": null,
                  "key": null,
                  "nodeType": "class",
                  "props": Object {
                    "children": Array [
                      "R$ ",
                      "bar",
                    ],
                    "hoverable": false,
                    "style": Object {
                      "textAlign": "right",
                    },
                  },
                  "ref": null,
                  "rendered": Array [
                    "R$ ",
                    "bar",
                  ],
                  "type": [Function],
                },
              ],
              "type": [Function],
            },
          ],
          "type": [Function],
        },
      ],
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
